src/bsoncxx/types.hpp

```diff
@@ -31,36 +31,15 @@ BSONCXX_INLINE_NAMESPACE_BEGIN
 /// An enumeration of each BSON type.
 ///
 enum class type : std::uint8_t {
-    k_double = 0x01,
-    k_utf8 = 0x02,
-    k_document = 0x03,
-    k_array = 0x04,
-    k_binary = 0x05,
-    k_undefined = 0x06,
-    k_oid = 0x07,
-    k_bool = 0x08,
-    k_date = 0x09,
-    k_null = 0x0A,
-    k_regex = 0x0B,
-    k_dbpointer = 0x0C,
-    k_code = 0x0D,
-    k_symbol = 0x0E,
-    k_codewscope = 0x0F,
-    k_int32 = 0x10,
-    k_timestamp = 0x11,
-    k_int64 = 0x12,
-    k_maxkey = 0x7F,
-    k_minkey = 0xFF
+#define BSONCXX_ENUM(name, val) k_##name = val,
+#include <bsoncxx/enums/type.hpp>
```

Comment:
I'm good with it too. Please do add a NOTE so that users are not utterly baffled. I think something like:

Please see the binary_sub_type header for the values that will be used here. The generated enumeration values are of the form k_oid, k_regex, etc.

And do the same NOTE for the other xmacro below.